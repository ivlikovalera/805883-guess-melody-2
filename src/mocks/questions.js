export const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `rock`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `pop`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `jazz`,
      },
      {
        src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `singer`,
    song: {
      singer: `Sting`,
      src: `https://soundtimes.ru/images/mp3/Shape%20Of%20My%20Heart.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        singer: `Sting`,
      },
      {
        picture: `http://placehold.it/134x134`,
        singer: `Eminem`,
      },
      {
        picture: `http://placehold.it/134x134`,
        singer: `Pink`,
      },
    ],
  },
];
