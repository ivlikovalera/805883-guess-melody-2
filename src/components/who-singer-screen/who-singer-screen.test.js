import React from 'react';
import renderer from 'react-test-renderer';
import WhoSingerScreen from './who-singer-screen.jsx';
import {questions} from '../../mocks/questions.js';

it(`WhoSingerScreen correctly renders`, () => {
  const singerScreen = renderer
  .create(<WhoSingerScreen
    question={questions[1]}
    screenIndex={0}
    onAnswer={() => {}}
  />)
  .toJSON();

  expect(singerScreen).toMatchSnapshot();
});
