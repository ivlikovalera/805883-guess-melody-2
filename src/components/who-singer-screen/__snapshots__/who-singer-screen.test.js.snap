// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhoSingerScreen correctly renders 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="false"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="Sting"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Sting
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="false"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Eminem"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Eminem
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="false"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Pink"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Pink
        </label>
      </div>
      ;
    </form>
  </section>
</section>
`;
